src/main/java/decodes/tsdb/algo/AddToPrevious.java

```diff
@@ -1,82 +1,51 @@
 package decodes.tsdb.algo;
 
-import java.util.Date;
-
 import opendcs.dai.TimeSeriesDAI;
 
-import ilex.var.NamedVariableList;
 import ilex.var.NamedVariable;
 import ilex.var.TimedVariable;
-import decodes.tsdb.BadTimeSeriesException;
 import decodes.tsdb.CTimeSeries;
-import decodes.tsdb.DbAlgorithmExecutive;
 import decodes.tsdb.DbCompException;
-import decodes.tsdb.DbIoException;
 import decodes.tsdb.ParmRef;
-import decodes.tsdb.VarFlags;
-
-//AW:IMPORTS
-// Place an import statements you need here.
-//AW:IMPORTS_END
-
-//AW:JAVADOC
-/**
-Adds the current value to the previous value in the database
-and outputs the sum. Works on any time-series, any interval.
-This algorithm does assume that you are calling it with a 
-series of contiguous values, like you would get out of a DCP
-message.
- */
-//AW:JAVADOC_END
-public class AddToPrevious
-	extends decodes.tsdb.algo.AW_AlgorithmBase
+import org.opendcs.annotations.algorithm.Algorithm;
+import org.opendcs.annotations.algorithm.Input;
+import org.opendcs.annotations.algorithm.Output;
+
+@Algorithm(
+		description = "Adds the current value to the previous value in the database\n" +
+				"\t\tand outputs the sum. Works on any time-series, any interval.\n" +
+				"\t\tThis algorithm does assume that you are calling it with a \n" +
+				"\t\tseries of contiguous values, like you would get out of a DCP\n" +
+				"\t\tmessage.")
+
+public class AddToPrevious extends decodes.tsdb.algo.AW_AlgorithmBase
 {
-//AW:INPUTS
-	public double input;	//AW:TYPECODE=i
-	String _inputNames[] = { "input" };
-//AW:INPUTS_END
-
-//AW:LOCALVARS
-	// Enter any local class variables needed by the algorithm.
+	@Input
+	public double input;
 	private double prevVal = 0.0;
 	private boolean justStarted = true;
-//AW:LOCALVARS_END
 
-//AW:OUTPUTS
+	@Output
 	public NamedVariable output = new NamedVariable("output", 0);
-	String _outputNames[] = { "output" };
-//AW:OUTPUTS_END
-
-//AW:PROPERTIES
-	String _propertyNames[] = { };
-//AW:PROPERTIES_END
 
 	// Allow javac to generate a no-args constructor.
 
 	/**
 	 * Algorithm-specific initialization provided by the subclass.
 	 */
-	protected void initAWAlgorithm( )
-		throws DbCompException
+	protected void initAWAlgorithm()
+			throws DbCompException
 	{
-//AW:INIT
 		_awAlgoType = AWAlgoType.TIME_SLICE;
-//AW:INIT_END
-
-//AW:USERINIT
-		// Code here will be run once, after the algorithm object is created.
-//AW:USERINIT_END
 	}
 	
 	/**
 	 * This method is called once before iterating all time slices.
 	 */
 	protected void beforeTimeSlices()
```

Comment:
If you feel you have the time, beforeTimeSlices, doAWTimeSlice, and afterTimeSlice should also have the `@Override` annotation.

Not huge deal, but it can save some trouble at compile time.